apps/src/blockly/TESTING.md

```diff
@@ -0,0 +1,79 @@
+- Click on the question mark icon of the unused code's svg frame. Check that a callout is displayed and that callout is removed when you click on X.
+
+# Dance Lab
+Create a new Dance Lab project: [local](http://localhost-studio.code.org:3000/projects/dance/new) [production](https://studio.code.org/projects/dance/new)
+- Check blocks that contain different types of fields such as dropdown menus.
+- Check blocks that contain multiple fields, for example,  '_alternate every_measures_between_and_'.
+- 'set background color' block should have a field that when click displays a 7x10 color grid while the 'set all tint' block's grid is 3x3.
+- Check functions. Compare with function definition blocks on production.
+    Once you create a drag out a function definition block, a 'do something' block should appear in Toolbox under 'Functions'. Once the function definition is removed from workspace, the 'do something' block should also be removed.
+- Go to this Dance Party level: [local](http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/1) [production](https://studio.code.org/s/dance-2019/lessons/1/levels/1).
+Check the block in the toolbox, in particular, that the dropdown field for type of dancer has only 2 options. Then click and drag a block to workspace and check the two dropdown menus again to see if the options and value are correctly displayed.
+
+# Flappy
+Create a new Flappy project: [local](http://localhost-studio.code.org:3000/projects/flappy/new) [production](https://studio.code.org/projects/flappy/new)
+- Event blocks, for example, 'when hit the ground', have fixed x/y position for blocks on workspace.
+- No shadow blocks are included in this lab.
+
+# Bounce and related skins
+Create a new Bounce project: [local_bounce](http://localhost-studio.code.org:3000/projects/bounce/new) [production_bounce](https://studio.code.org/projects/bounce/new)
+
+Create a new Basketball project: [local_basketball](http://localhost-studio.code.org:3000/projects/basketball/new) [production_basketball](https://studio.code.org/projects/basketball/new)
+
+Create a new Sports project: [local_sports](http://localhost-studio.code.org:3000/projects/sports/new) [production_sports](https://studio.code.org/projects/sports/new)
+- Has 'skins' that are very similar - Sports and Basketball projects.
+- In Basketball, the 'set hand' block has 3 hands.
+- In Sports, the 'set player' block has 3 hands + 2 items.
+
+# Poetry
+Create a new Poetry project: [local_poetry](http://localhost-studio.code.org:3000/projects/poetry/new) [production_poetry](https://studio.code.org/projects/poetry/new)
+
+Create a new Poetry HOC project: [local_poetry_hoc](http://localhost-studio.code.org:3000/projects/poetry/new) [production_poetry_hoc](https://studio.code.org/projects/poetry/new)
+- Check 2 versions of Poetry lab - poetry and poetry_hoc
+- Check shadow blocks, for example, 'set title' block. Most blocks that accept text have shadow blocks.
```

Comment:
You will have to remove another "real" block that is covering the shadow block in order to see it.